src/utils.ts

```diff
@@ -34,3 +36,28 @@ export function getFillOpacity(selected: boolean, highlight: boolean, hasSelecti
 
     return DefaultOpacity;
 }
+
+export function getLegendFillOpacity(
+    selected: boolean,
+    hasSelection: boolean,
+    isHCM: boolean): number {
+
+    if ((hasSelection && !selected) && isHCM) {
+        return DimmedOpacity;
+    }
+
+    return DefaultOpacity;
+}
+
+export function getLegendFill(
+    selected: boolean,
+    hasSelection: boolean,
+    defaultColor: string,
+    isHCM: boolean): string {
```

Comment:
`isHCM` is a [bad naming](https://github.com/microsoft/PowerBI-visuals-AsterPlot/pull/90/files#r1793304169)

Reply:
Renamed.